### PR TITLE
Revised Sign-up experience

### PR DESCRIPTION
**Pull request by erimerkin**
This PR's details are explained in #418. The purpose of the changes are creating a revised sign-up experience for users. 

2 new pages using EditUserInfo component are created for updating user information. The first page is shown automatically after sign-up and the second page handles the information editing on `/profile/settings`.

While implementing these details 6 new common components were implemented:
- LoadingButton
- ImageUploader
- EditUserInfo
- CountrySelect
- CustomOutlinedInput
- GenericCardLayout

#### For future needed changes:
- Implement an image component to automatically fetch images from the API with their IDs

### CODE REVIEW

File: frontend/src/pages/Authentication/SignInPage.js [hunk 1]
```diff
@@ -38,107 +33,99 @@ function SignInForm(props) {
...
-      .then((responseText) => {
-        if (responseText) {
-          saveToken(responseText);
-          navigate('/');
-        } else {
-          formik.setFieldValue("password", "");
-          setError("Your email or password is incorrect. Please try again.");
+      // Here is how we will make a POST request in the backend.
+      // This section is left out since the backend is not ready
+      // yet.
```
Comment (CahidArda) [inline frontend/src/pages/Authentication/SignInPage.js:47]: This comment can be removed.

File: frontend/src/pages/ProfilePage/SettingsPage.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                setError(error.message);
+            }
+            );
+    }, [token]
+    )
+
+
+
+    return (
+        console.log(userInfo),
```
Comment (CahidArda) [inline frontend/src/pages/ProfilePage/SettingsPage.js:58]: We can remove console.log

**Review by CahidArda — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Authentication/SignInPage.js (hunk 1 above)
Comment (erimerkin) [reply frontend/src/pages/Authentication/SignInPage.js]: Removed in b04be1a.

File: frontend/src/pages/ProfilePage/SettingsPage.js (hunk 2 above)
Comment (erimerkin) [reply frontend/src/pages/ProfilePage/SettingsPage.js]: Removed in b04be1a.

**Comment by erimerkin** [comment]
@CahidArda  I have fixed the parts you requested in commit b04be1a.

**Review by CahidArda — APPROVED** [review]
LGTM!

**Pull request merged**